### PR TITLE
feat: unconfirmed user login error fix

### PR DESCRIPTION
**Pull request by YazeedLoonat**
## Issue Overview

This PR addresses [#issue](https://github.com/bloom-housing/bloom/issues/3912)

- [x] This change addresses the issue in full

## Description
- This makes it so an unconfirmed user on the public site attempting to login is prompted to resend the confirmation email

- this moves the password validity check earlier during login

- public users can no longer log into partner site

- adds spanish translations to seeding script

## How Can This Be Tested/Reviewed?
- create a user on the public site (do not confirm them) attempt to log in, you should now be prompted to resend confirmation email

- as a public only user you should no longer be able to log into the partner site (note locally you will be auto logged in if you are logged into the public site

## Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have added QA notes to the issue with applicable URLs
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have assigned reviewers
- [x] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [x] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [x] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

File: api/prisma/seed-staging.ts [hunk 1]
```diff
@@ -92,6 +93,9 @@ export const stagingSeed = async (
   await prismaClient.translations.create({
     data: translationFactory(jurisdiction.id, jurisdiction.name),
   });
+  await prismaClient.translations.create({
+    data: translationFactory(undefined, undefined, LanguagesEnum.es),
```
Comment (mcgarrye) [inline api/prisma/seed-staging.ts:97]: why no jurisdiction for this translation?

Comment (YazeedLoonat) [reply api/prisma/seed-staging.ts:97]: no jurisdiction just means that its the generic translation set on the server if that makes sense

**Comment by YazeedLoonat** [comment]
Nice catch @cade-exygy should be fixed now

**Review by cade-exygy — COMMENTED** [review]
Screenshot 2024-03-14 at 11 07 21 AM
found another one!

**Comment by cade-exygy** [comment]
Approved!

**Review by cade-exygy — APPROVED** [review]

**Comment by ColinBuyck** [comment]
Not blocking but was surprised to see no title on this error messaging
Screenshot 2024-03-14 at 1 59 50 PM

**Comment by YazeedLoonat** [comment]
@ColinBuyck 

yeaaa that pop looks a bit out of place, and could use some touch ups but its what we had for users that were attempting to login in unconfirmed 

I did show screenshots to design and they okay'd it for now, but yeah should probably make that better

File: shared-helpers/src/auth/AuthContext.ts [hunk 2]
```diff
@@ -223,16 +224,25 @@ export const AuthProvider: FunctionComponent<React.PropsWithChildren> = ({ child
       email,
       password,
       mfaCode: string | undefined = undefined,
-      mfaType: MfaType | undefined = undefined
+      mfaType: MfaType | undefined = undefined,
+      forPartners: boolean | undefined = undefined
```
Comment (ColinBuyck) [inline shared-helpers/src/auth/AuthContext.ts:228]: Why do we need this forPartners piece?

**Review by ColinBuyck — APPROVED** [review]
LGTM 🪀 one question for my understanding

File: shared-helpers/src/auth/AuthContext.ts (hunk 2 above)
Comment (YazeedLoonat) [reply shared-helpers/src/auth/AuthContext.ts:228]: the backend doesn't really know if the user logging in is logging into public or partners

so where we do the login calls we make sure the user is able to login to partners

**Pull request merged**